main.cpp

```diff
@@ -47,20 +125,28 @@ HANDLE WINAPI HookOpenEvent(DWORD dwDesiredAccess, BOOL bInheritHandle, LPCWSTR
 BOOL APIENTRY DllMain(HANDLE hModule, DWORD ul_reason_for_call, LPVOID lpReserved)
 {
 	ReviveModule = hModule;
```

Comment:
Perhaps if we move this line to `DLL_PROCESS_ATTACH` then maybe it won't be overwritten by an invalid handle.